### PR TITLE
Added linting rule to block text-transform

### LINKED ISSUE #15: Add a linting rule to block additional usage of `text-transform`
Adding text-transform: uppercase (or others) creates added friction for translated content.

While it may not be critical we want to ensure that our stylelint config makes it simple to make things translatable and usable.

_Originally posted by @lb- in https://github.com/wagtail/wagtail/issues/7624#issuecomment-1058879006_

### PR DESCRIPTION
**Pull request by PaarthAgarwal**
I have added added text-transform: uppercase to declaration-property-value-disallowed-list. Fixes #15 .

### CODE REVIEW

**Comment by lb-** [comment]
Hey. I think we would like to block the usage of `text-transform` completely.

**Comment by PaarthAgarwal** [comment]
> Hey. I think we would like to block the usage of `text-transform` completely.

I'm new to stylelint. Can you please me guide how to achieve it?

**Comment by PaarthAgarwal** [comment]
> Hey. I think we would like to block the usage of `text-transform` completely.

I think I got your point now. I have made some changes accordingly, please check

File: index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ module.exports = {
     'declaration-property-value-disallowed-list': [
       { '/^border/': ['none'] },
       { severity: 'error' },
+      { 'text-transform': 'uppercase' },
```
Comment (lb-) [inline index.js:26]: I think this line can be removed now, as we are disallowing the property

Comment (PaarthAgarwal) [reply index.js]: done

**Review by thibaudcolas — APPROVED** [review]
Thank you both, looking great!

**Pull request merged**